ckan/lib/uploader.py

```diff
@@ -218,22 +218,22 @@ def upload(self, id, max_size=10):
                     raise
             tmp_filepath = filepath + '~'
             with open(tmp_filepath, 'wb+') as output_file:
-                with self.upload_file:
-                    self.upload_file.seek(0)
-                    current_size = 0
-                    while True:
-                        current_size = current_size + 1
-                        # MB chunks
-                        data = self.upload_file.read(2 ** 20)
-
-                        if not data:
-                            break
-                        output_file.write(data)
-                        if current_size > max_size:
-                            os.remove(tmp_filepath)
-                            raise logic.ValidationError(
-                                {'upload': ['File upload too large']}
-                            )
+                self.upload_file.seek(0)
```

Comment:
Better to change `with` to `try ... finally`. Otherwise it would still be possible to leak file handles and therefore disk space (eg when an upload exceeds `max_size`).

Reply:
I wonder why this isn't a problem in the upstream fix. Or maybe it is.

Reply:
I asked a question on the upstream PR (which _still_ isn't merged yet).